src/resdk/tables/variant.py

```diff
@@ -79,90 +80,185 @@ class VariantTables(BaseTables):
+            raise ValueError("It is not allowed to change geneset value.")
+
+        if value is None:
+            return
+
+        # If id / slug of a geneset is given, get it from the Resolwe server
+        if isinstance(value, (int, str)):
+            gs = self.resolwe.geneset.get(value)
+            value = gs.genes
+        elif isinstance(value, Geneset):
+            value = value.genes
+
+        if isinstance(value, (list, set, tuple, pd.Series)):
+            self._geneset = set(value)
         else:
-            # Check for heterogeneous mutations only if mutations is not specified
-            self.check_heterogeneous_mutations()
+            raise ValueError(f'Unsupported type of "geneset" input: {value}.')
+
+    @property
+    @lru_cache()
+    def variants(self) -> pd.DataFrame:
+        """Get variants table.
 
-    def check_heterogeneous_mutations(self):
-        """Ensure variants are selected for the same mutations in all samples."""
-        mutations = {str(d.input["mutations"]) for d in self._data}
+        There are 4 possible values:
+
+            - 0 - wild-type, no variant
+            - 1 - heterozygous mutation
+            - 2 - homozygous mutation
+            - NaN - QC filters are failing - mutation status is unreliable
+
+        """
+        df = self._load_fetch(self.VARIANTS)
+
+        # Variants that are not reported (NaN) were not detected:
+        # they are wild type.
+        df = df.fillna(0)
+
+        if self.filtering:
+            # Keep values in case .filter == PASS or .variants == 0
+            df = df.where((self.filter == "PASS") | (df == 0), other=np.nan)
+
+        return df
+
+    @property
+    @lru_cache()
+    def depth(self) -> pd.DataFrame:
+        """Get depth table."""
+        return self._load_fetch(self.DEPTH)
+
+    @property
+    @lru_cache()
+    def depth_a(self) -> pd.DataFrame:
+        """Get depth table for adenine."""
+        return self._load_fetch(self.DEPTH_A)
+
+    @property
+    @lru_cache()
+    def depth_c(self) -> pd.DataFrame:
+        """Get depth table for cytosine."""
+        return self._load_fetch(self.DEPTH_C)
+
+    @property
+    @lru_cache()
+    def depth_g(self) -> pd.DataFrame:
+        """Get depth table for guanine."""
+        return self._load_fetch(self.DEPTH_G)
+
+    @property
+    @lru_cache()
+    def depth_t(self) -> pd.DataFrame:
+        """Get depth table for thymine."""
+        return self._load_fetch(self.DEPTH_T)
+
+    # TODO: consider better name
+    @property
+    @lru_cache()
+    def filter(self) -> pd.DataFrame:
+        """Get filter table.
+
+        Values can be:
+
+            - PASS - Variant has passed filters:
+            - DP : Insufficient read depth
+            - QD: insufficient quality normalized by depth
+            - FS: insufficient phred-scaled p-value using Fisher's exact
+                test to detect strand bias
+            - SnpCluster: Variant is part of a cluster
+
```

Comment:
Maybe you can add specific filters used: FS > 30.0, QD < 2.0, DP < 10.0. So for example, if a variant has DP of 8, GATK will replace PASS with DP in FILTER column.